### DISCUSSION TITLE
Intro video for buildpacks

### DISCUSSION THREAD

**Discussion by yaelharel** [buildpacks/community#69] [Ideas]
I think it can be very useful to have a ~1 minute introduction video to what buildpacks are.
We can post this video in our website and share it in social media, and hopefully this way more people will get familiar with it.
An example of such video is [this intro video for zoom](https://www.youtube.com/watch?v=lfqo8whfL3g) but of course our video would be completely different and would probably include more animations.
Would love to get your thoughts about it.
Thanks!

**Comment by aemengo** [comment]
Really like this idea.

I'm thinking what's most important is the audience. 🤔 Plenty of these intro type videos are trying to sell you on a product by showcasing marquee features.

If our audience is, say contributors, maybe we want to show the moving parts of buildpacks (a map) so you that you know where to jump in to explore more.
If our audience is, say developers, maybe we want to show how buildpacks fits into your normal development loop.

**Reply by dfreilich** [reply]
I would assume that the main audience would be developers – I feel like for contributors, we would point them towards some of the conference presentation recordings

**Reply by jromero** [reply]
I agree that contributors should not be the primary targeted audience. It should be the potential end-users (app devs, operators, stack/buildpack authors).

**Reply by thephw** [reply]
A one minute intro video on buildpacks targeted for application developers would be dope. We've got a contractor that does that type of work if we put together a script. Spaceship can likely even pay for it. 💸

**Reply by jromero** [reply]
@thephw that sounds awesome! 🙇 

**The script should be the first actionable item anyone in the community can work on.** 

From my perspective, I think we should portray the project at aiming for separation of concerns at a high-level and therefore have a "multiple hats" scenario but don't know if that's too prescriptive.

We can introduce characters:
- "Jill is an app developer, she cares about building an app that is..."
- "Jack is an devops engineer, he wants to make sure that every application deployed has..."

Alternatively, leave it generic:
- Intro: "Cloud Native Buildpacks is a ...."
- "App Developers can leverage buildpacks to... "
- "Organizations can create buildpacks, rely on community maintained buildpacks, or subscribe to buildpack vendors"

**Reply by yaelharel** [reply]
I love the idea of characters. I think people will be able to understand better how they can use buildpacks depending on their role.

**Comment by yaelharel** [comment]
I scheduled a meeting for next Tuesday (4/6) - 2:30-3:30pm to brainstorm some ideas for the intro video.
Zoom link: https://VMware.zoom.us/j/92066692967?pwd=OGx5V05YeGxuT3Mydzd6MDRFTmJ2dz09&from=addon
Miro board: https://miro.com/app/board/o9J_lMa7hwU=/
Let me know if you would like to get a calendar invite.

**Reply by dwillist** [reply]
## Summary from above meeting
We used [this Miro board](https://miro.com/app/board/o9J_lMa7hwU=/) to guide our discussion

## Who is the audience:
We found there are likely three type of users we are going to target with this video:
- new to container building, maybe a little experience using docker.
- platform operators looking to use buildpacks to use buildpacks to enable tighter control over application builds and management of running applications.
- People interested in singular features, such as fast build, etc.

## Script type
We propose a couple models for different scripts. please check them out on the Miro board above. If you have any though/additional ideas feel free to add them!

## Content
Here we are trying to best answer the following to guide the content that should appear in this video
- what are buildpacks.
    - We have some attempts at answering this question concisely, three basic points we though we need are:
      - container building tool
      - general contract (anyone can write/implement)
      - ecosystem of other users / authors / platforms.
    -Any feedback / contribution on these would be greatly appreciated! 
- why use buildpacks
  - TODO next time! 

Again feel free to add to the miro board but please no deletions.

**Reply by yaelharel** [reply]
As you can see in the miro board, we had a brainstorming about the content of the video. We separated it into _“what?”_ and _“why?“_. Many ideas were added to the _“why?”_ section but only few were added to the _“what?”_ section.

We need your help! In one clear sentence:
**What is Cloud Native Buildpacks?**

Would love to get your answers in this thread.

(Related to [this slack thread](https://buildpacks.slack.com/archives/C94UJCNV6/p1617824448060000))

**Reply by natalieparellano** [reply]
Here is my best effort based on the content in the board and what makes sense in my own head:

What are buildpacks? Buildpacks are programs that build your code for you. They can provide dependencies, run compilation, configure runtime environment variables, and more! The Cloud Native Buildpacks project provides a specification for buildpacks to follow that ensures builds are fast and secure. A buildpack can be written by anyone, and shared with everyone! Google, Heroku/Salesforce, and the Paketo project provide buildpacks for a wide variety of language families, and you can discover even more by searching the buildpack registry.

How do buildpacks work? Buildpacks are run by a program called the lifecycle, which packages your application into a container image that can be run on any cloud. The Cloud Native Buildpacks ecosystem has tooling and integrations that make it easy to run the lifecycle wherever you need to - on your workstation, in your favorite CI/CD system, and on Kubernetes. 

Why use buildpacks? As an application developer, you can focus on what you do best - writing code - without having to worry about maintaining Dockerfiles. As a platform operator, you get more control over the build process by choosing the buildpacks that work best for your organization. You’ll also enable quick and easy updates and patching of vulnerabilities, reproducible images, Bill of Materials, and more!

[More]

**Reply by sampeinado** [reply]
A couple things to add to Natalie's comment: 
- Buildpacks scan your code locally to determine how to build the app, but they do NOT store or send any info about your app 
- Buildpacks are a way to create standardization amongst application container builds, the value they bring is directly proportional to the number of applications/microservices you are managing  
- Its important to mention Docker at the start, to connect to something most people have heard of, because many people barely know what containers are
- it might be helpful to compare CNB lifecycle to an API and buildpacks to be the "content" options that the API can serve to developers as deps and OSes for their apps container environment

**Comment by yaelharel** [comment]
The next session is going to be next Wednesday (4/14) - 10:30-11:30 EST.
Miro board: https://miro.com/app/board/o9J_lMa7hwU=/
Slack thread: https://buildpacks.slack.com/archives/C94UJCNV6/p1617824448060000
Zoom link: https://VMware.zoom.us/j/95902027339?pwd=cldGQ3ovZFVtLytKWHFJeUFUSWdWZz09&from=addon
Please let me know if you would like to get a calendar invite.

**Reply by thephw** [reply]
This looks great! I'm on vacation next week, but toss me an invite if I'm awake I can pop in :-) 

patrick.wiseman@spaceship.run

**Comment by thephw** [comment]
@hunterbecton We might do this next.

**Comment by yaelharel** [comment]
Please see [this draft script](https://docs.google.com/document/d/1jKNU4sLhm0aQ6eU_JTwRhsqefnPBsrMXt5cPKcil-Ec/edit?usp=sharing) that @thephw wrote and feel free to comment/edit.
@thephw, thanks a lot for putting this together! It's great!

**Reply by thephw** [reply]
@natalieparellano Merged in a lot of that feedback and left a couple things open for discussion. I don't have any super-strong opinions, but wanted to give some context as to what I was thinking about.

**Comment by yaelharel** [comment]
Thank you for everyone who took the time and wrote a script!
I created one combined draft script from all of the suggestions.
Please review it [here](https://docs.google.com/document/d/1jKNU4sLhm0aQ6eU_JTwRhsqefnPBsrMXt5cPKcil-Ec/edit?usp=sharing) under the "Combined draft" section (all of the suggestions are in the second page of the document).
We would love to get your comments.
Thank you!

**Comment by yaelharel** [comment]
I've opened [this RFC](https://github.com/buildpacks/rfcs/pull/161) to finalize the script.
